app/backend/src/clj/teet/environment.clj

```diff
@@ -299,17 +299,19 @@
 
 (defn datomic-connection
   "Returns thread bound connection or creates a new one."
-  []
-  (or *connection*
-      (let [db (db-name)
-            client (datomic-client)
-            db-status (ensure-database client db)
-            conn (d/connect client {:db-name db})]
-        (log/info "Using database: " db db-status)
-        (when-not @db-migrated?
-          (migrate conn @schema)
-          (reset! db-migrated? true))
-        conn)))
+  ([]
+   (datomic-connection [(db-name)]))
+  ([database-name]
+   (or *connection*
```

Comment:
what if thread bound `*connection*` is to a database that has a different name?

Reply:
I can't imagine the situation. Reasonbly to assume that the caller of this function with database-name param must have checked the name validity..